Purchases/SwiftSources/LocalReceiptParsing/Builders/ASN1ContainerBuilder.swift

```diff
@@ -0,0 +1,80 @@
+//
+// Created by Andrés Boedo on 7/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+struct ASN1ContainerBuilder {
+
+    func build(fromPayload payload: ArraySlice<UInt8>) throws -> ASN1Container {
```

Comment:
all builders have a single public method, `build` that takes an `ArraySlice` or an `ASN1Container`, and return the built object. 

quick note on `ArraySlice` - it's a reference to the sliced array, but not a copy, making it very efficient to pass around. 
Note that passing an `Array` around would also be efficient because Swift does copy-on-write for collections, so they wouldn't get copied in our case. 
However, `ArraySlice` provides some nice convenience methods for slicing further.